### PR TITLE
Add helpful error when failing to read compressed result files

### LINKED ISSUE #409: Bug located in ...pymapdl_reader
### 🤓 Before submitting the issue

- [X] I have visited the [Troubleshooting section](https://mapdl.docs.pyansys.com/troubleshoot/index.html).
- [X] I have searched among the [existing issues](https://github.com/ansys/pymapdl/issues?q=).
- [X] I am using a [Python virtual environment](https://docs.python.org/3/library/venv.html).
- [X] I am using the [latest version of PyMAPDL](https://github.com/ansys/pymapdl/releases) and its dependencies (fully updated virtual environment). You can update them using ``pip install --upgrade --upgrade-strategy eager ansys-mapdl-core`` in your activated virtual environment.

### 🔍 Description of the bug

Eexception thrown out sometimes is 
"  longint = struct.pack(">I", inth) + struct.pack(">I", intl) struct.error: required argument is not an integer" while reading rst file by code results = pymapdl_reader.read_binary("job.rst") . 

or sometimes is
 "  File "C:\Users\86189\Desktop\JOB\Code\pyAnsysDemo\CZ_FATRES\_ansysReader.py", line 6, in <module>
    result = pymapdl_reader.read_binary(r"job.rst")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\software\Python3.12\Lib\site-packages\ansys\mapdl\reader\common.py", line 173, in read_binary
    result = Result(filename, read_mesh=False, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\software\Python3.12\Lib\site-packages\ansys\mapdl\reader\rst.py", line 145, in __init__
    self._resultheader = self._read_result_header()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\software\Python3.12\Lib\site-packages\ansys\mapdl\reader\rst.py", line 235, in _read_result_header
    header = parse_header(self.read_record(103), result_header_keys)
                          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\software\Python3.12\Lib\site-packages\ansys\mapdl\reader\common.py", line 110, in read_record
    record = self._cfile.read_record(pointer, return_bufsize)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "ansys\\mapdl\\reader\\cython\\_binary_reader.pyx", line 251, in ansys.mapdl.reader._binary_reader.AnsysFile.read_record
  File "ansys\\mapdl\\reader\\cython\\_binary_reader.pyx", line 367, in ansys.mapdl.reader._binary_reader.wrap_array
  File "ansys\\mapdl\\reader\\cython\\_binary_reader.pyx", line 158, in ansys.mapdl.reader._binary_reader.ArrayWrapper.__array__
ValueError: negative dimensions are not allowed"

The rst file can be open by both ansys workbench and HyperView. So it seems a bug for pymapdl reader.

### 🕵️ Steps To Reproduce

```Python
from ansys.mapdl import reader as pymapdl_reader
results = pymapdl_reader.read_binary("job.rst")
I don't know how to upload the rst file here, anyone who know it please tell me.
```

### 💻 Which Operating System are you using?

Windows

### 🐍 Which Python version are you using?

3.12

### 📝 PyMAPDL Report

<details><summary>Show the Report!</summary>

PyMAPDL Software and Environment Report

Packages Requirements
*********************

Core packages
-------------
ansys.mapdl.core    : 0.67.0
numpy               : 1.26.4
platformdirs        : 4.2.0
scipy               : 1.13.0
grpc                : Package not found
ansys.api.mapdl.v0  : Package not found
ansys.mapdl.reader  : 0.53.0
google.protobuf     : Package not found

Optional packages
-----------------
matplotlib          : 3.8.4
pyvista             : 0.43.5
pyiges              : 0.3.1
tqdm                : 4.66.2

Ansys Installation
******************
Unable to locate any Ansys installations

Ansys Environment Variables
***************************
</details>

### 📝 Installed packages

ansys-api-fluent==0.3.12
ansys-api-mapdl==0.5.1
ansys-api-mechanical==0.1.0
ansys-api-meshing-prime==0.1.1
ansys-api-platform-instancemanagement==1.0.0
ansys-api-pyensight==0.3.0
ansys-api-sherlock==0.1.14
ansys-api-systemcoupling==0.1.0
ansys-corba==0.1.1
ansys-dpf-composites==0.2.0
ansys-dpf-core==0.8.1
ansys-dpf-gate==0.3.1
ansys-dpf-gatebin==0.4.1
ansys-dpf-post==0.4.0
ansys-dynamicreporting-core==0.4.0
ansys-fluent-core==0.14.0
ansys-grantami-bomanalytics==1.1.3
ansys-grantami-bomanalytics-openapi==1.0.0
ansys-grantami-recordlists==1.0.0
ansys-grantami-serverapi-openapi==1.0.0
ansys-grpc-dpf==0.8.1
ansys-mapdl-core==0.64.1
ansys-mapdl-reader==0.53.0
ansys-math-core==0.1.1
ansys-mechanical-core==0.8.0
ansys-meshing-prime==0.4.1
ansys-motorcad-core==0.1.3
ansys-openapi-common==1.2.1
ansys-optislang-core==0.2.1
ansys-platform-instancemanagement==1.1.1
ansys-pyensight-core==0.5.3
ansys-pythonnet==3.1.0rc1
ansys-seascape==0.2.0
ansys-sherlock-core==0.2.0
ansys-systemcoupling-core==0.1.3

### 📝 Logger output file

<details><summary>Show the logger output file.</summary>

```text

===============================================================================
       NEW SESSION - 04/05/2024, 14:32:40
===============================================================================
LEVEL - INSTANCE NAME - MODULE - FUNCTION - MESSAGE
DEBUG - pymapdl_global -  launcher - launch_mapdl - No IP address was supplied. Using the default IP address: 127.0.0.1
DEBUG - pymapdl_global -  launcher - launch_mapdl - Using default port 50052
DEBUG - pymapdl_global -  launcher - launch_mapdl - Using 'start_instance' equal to True
DEBUG - pymapdl_global -  launcher - launch_mapdl - Using default executable.

```
</details>

### PR DESCRIPTION
**Pull request by akaszynski**
Resolve #409 by adding a helpful error when failing to read compressed result files, suggesting the user disable compression with `/FCOMP,RST,0`.

### CODE REVIEW

File: ansys/mapdl/reader/rst.py [hunk 1]
```diff
@@ -233,7 +233,16 @@ def _read_result_header(self):
         standard_header = read_standard_header(self.filename)
 
         # Read .RST FILE HEADER
-        header = parse_header(self.read_record(103), result_header_keys)
+        try:
+            record = self.read_record(103)
+            if record.size > 80:
+                raise RuntimeError("Invalid record size. Should be 80 entries")
+            header = parse_header(record, result_header_keys)
+        except:
```
Comment (germa89) [inline ansys/mapdl/reader/rst.py:241]: Do we know the expected exception? We should avoid bare exceptions as much as possible.

File: ansys/mapdl/reader/rst.py [hunk 2]
```diff
@@ -233,7 +233,16 @@ def _read_result_header(self):
...
+            raise RuntimeError(
```
Comment (germa89) [inline ansys/mapdl/reader/rst.py:242]: Also I rather you re-raise the initial error, just pre-pending the new message to the previous error message, so you do not hide potentially useful information.

**Review by germa89 — CHANGES_REQUESTED** [review]

**Review by germa89 — APPROVED** [review]

File: ansys/mapdl/reader/rst.py (hunk 1 above)
Comment (akaszynski) [reply ansys/mapdl/reader/rst.py]: Cleaned up the error in [7a3a5b8](https://github.com/ansys/pymapdl-reader/pull/474/commits/7a3a5b81bd4ba4b3521732b8003d52815365cfdb).

File: ansys/mapdl/reader/rst.py (hunk 2 above)
Comment (akaszynski) [reply ansys/mapdl/reader/rst.py:238]: Agreed. Opted for checking the header size rather than placing this in a try/except.

**Comment by RobPasMue** [comment]
Hi @akaszynski - we'll need to do a new minor release once this is merged. The numpy 2.0 change will most probably break things for certain users.

**Comment by germa89** [comment]
It looks good to me with the new changes.

**Pull request merged**